welkin/models/base.py

```diff
@@ -45,6 +45,9 @@ def __getattr__(self, name):
     def __setattr__(self, name, value):
         super().__setitem__(name, value)
 
+    def __delattr__(self, name):
```

Comment:
Added this as a convenience. Previously, if one tried to `del` an attribute of this class, an `AttributeError` would be raised. The only way to actually delete an attribute was with dictionary syntax. Now both are possible.

```python
r = Resource(id="123")
del r["id"]
del r.id  # Previously not possible
```